discovery-provider/src/premium_content/premium_content_access_checker.py

```diff
@@ -0,0 +1,106 @@
+import json
+import logging
+from typing import Dict, Optional, Tuple, cast
+
+from src.models.tracks.track import Track
+from src.premium_content.helpers import does_user_have_nft_collection
+from src.premium_content.types import PremiumContentType
+from src.utils import db_session
+
+logger = logging.getLogger(__name__)
+
+
+class PremiumContentAccessChecker:
+    # check if content is premium
+    # - if not, then user has access
+    # - if so, check whether user fulfills the conditions
+    #
+    # Returns a tuple of (bool, bool) -> (track is premium, user has access)
+    #
+    # Note: premium content id for type should exist, but just in case it does not,
+    # we return true for user access so that (non-existent) track does not change
+    # existing flow of the function caller.
+    def check_access(
+        self,
+        user_id: int,
+        premium_content_id: int,
+        premium_content_type: PremiumContentType,
+    ) -> Tuple[bool, bool]:
+        # for now, we only allow tracks to be premium
+        # premium playlists will come later
+        if premium_content_type != "track":
+            return False, True
+
+        (
+            is_premium,
+            premium_conditions,
+            content_owner_id,
+        ) = self._is_content_premium(premium_content_id)
+
+        if not is_premium:
+            # premium_conditions should always be null here as it makes
+            # no sense to have a non-premium track with conditions
+            if premium_conditions:
+                logger.warn(
+                    f"premium_content_access_checker.py | _aggregate_conditions | non-premium content with id {premium_content_id} and type {premium_content_type} has premium conditions."
```

Comment:
maybe premium_conditions should not be nullable and then we can just check premium_conditions if it actually is_premium and leave it an empty JSON otherwise? whichever seems simpler to you.